samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,401 @@
+	{ "cellular_timeout", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_TIMEOUT },
+	{ "cellular_service", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_SERVICE },
+	{ "wifi_timeout", required_argument, 0, LOCATION_SHELL_OPT_WIFI_TIMEOUT },
+	{ "wifi_service", required_argument, 0, LOCATION_SHELL_OPT_WIFI_SERVICE },
+	{ 0, 0, 0, 0 }
+};
+
+/******************************************************************************/
+
+static void location_shell_print_usage(const struct shell *shell,
+				       struct location_shell_cmd_args *loc_cmd_args)
+{
+	switch (loc_cmd_args->command) {
+	case LOCATION_CMD_GET:
+		mosh_print_no_format(location_get_usage_str);
+		break;
+
+	default:
+		mosh_print_no_format(location_usage_str);
+		break;
+	}
+}
+
+#define MOSH_LOC_SERVICE_NONE 0xFF
+
+static enum location_service location_shell_string_to_service(const char *service_str)
+{
+	enum location_service service = MOSH_LOC_SERVICE_NONE;
+
+	if (strcmp(service_str, "any") == 0) {
+		service = LOCATION_SERVICE_ANY;
+	} else if (strcmp(service_str, "nrf") == 0) {
+		service = LOCATION_SERVICE_NRF_CLOUD;
+	} else if (strcmp(service_str, "here") == 0) {
+		service = LOCATION_SERVICE_HERE;
+	} else if (strcmp(service_str, "skyhook") == 0) {
+		service = LOCATION_SERVICE_SKYHOOK;
+	} else if (strcmp(service_str, "polte") == 0) {
+		service = LOCATION_SERVICE_POLTE;
+	}
+
+	return service;
+}
+
+/******************************************************************************/
+
+void location_ctrl_event_handler(const struct location_event_data *event_data)
+{
+	char gmaps_str[512];
+
+	switch (event_data->id) {
+	case LOCATION_EVT_LOCATION:
+		mosh_print("Location:");
+		mosh_print(
+			"  used method: %s (%d)",
+			location_method_str(event_data->location.method),
+			event_data->location.method);
+		mosh_print("  latitude: %.06f", event_data->location.latitude);
+		mosh_print("  longitude: %.06f", event_data->location.longitude);
+		mosh_print("  accuracy: %.01f m", event_data->location.accuracy);
+		if (event_data->location.datetime.valid) {
+			mosh_print(
+				"  date: %04d-%02d-%02d",
+				event_data->location.datetime.year,
+				event_data->location.datetime.month,
+				event_data->location.datetime.day);
+			mosh_print(
+				"  time: %02d:%02d:%02d.%03d UTC",
+				event_data->location.datetime.hour,
+				event_data->location.datetime.minute,
+				event_data->location.datetime.second,
+				event_data->location.datetime.ms);
+		}
+		sprintf(gmaps_str, "  Google maps URL: https://maps.google.com/?q=%f,%f",
+			event_data->location.latitude, event_data->location.longitude);
+		mosh_print("%s", gmaps_str);
+		break;
+
+	case LOCATION_EVT_TIMEOUT:
+		mosh_error("Timeout happened when getting a location");
+		break;
+
+	case LOCATION_EVT_ERROR:
+		mosh_error("An error happened during getting a location");
+		break;
+	case LOCATION_EVT_GNSS_ASSISTANCE_REQUEST:
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
```

Comment:
I think this flagging could be removed. If we think about modem_shell as a test application, it would be good to print the event data regardless of the CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL flag. This way an unexpected LOCATION_EVT_GNSS_ASSISTANCE_REQUEST event from the Location library would also get printed.

Reply:
event_data->request is flagged out of the API if `CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL` is not configured.